### PR TITLE
Modify the handling of exit code in lpython.cpp

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Closes: #1316

### CODE REVIEW

File: src/bin/lpython.cpp [hunk 1]
```diff
@@ -1657,6 +1657,8 @@ int main(int argc, char *argv[])
                 if (err != 0) {
                     if (0 < err && err < 256) {
                         return err;
+                    } else if (err % 256 == 0) {
+                        return err / 256;
```
Comment (certik) [inline src/bin/lpython.cpp:1661]: What does this do exactly? Let's document it in a comment.

Comment (Thirumalai-Shaktivel) [reply src/bin/lpython.cpp:1661]: In Linux, the exit values are multiplied by 256
https://stackoverflow.com/a/3736358
unix.stackexchange.com/questions/394639/why-do-high-exit-codes-on-linux-shells-256-not-work-as-expected
So, when we do:
```py
quit(10)
```
```
>> echo $?
2560
```

File: src/bin/lpython.cpp [hunk 2]
```diff
@@ -1658,7 +1658,9 @@ int main(int argc, char *argv[])
                     if (0 < err && err < 256) {
                         return err;
                     } else {
-                        return 1;
+                        // https://stackoverflow.com/a/27117435/15913193
+                        // https://linux.die.net/man/3/system
+                        return WEXITSTATUS(err);
```
Comment (certik) [inline src/bin/lpython.cpp:1663]: This does not compile on Windows:
```
D:\a\lpython\lpython\lpython-0.8.0-14-gb2d00ab00\src\bin\lpython.cpp(1663): error C3861: 'WEXITSTATUS': identifier not found
```

Comment (Thirumalai-Shaktivel) [reply src/bin/lpython.cpp:1663]: I'm Sorry, @certik. I forgot the suggestions you gave on this.
Would this: 0e9995174cc1e3cdf8b4363f630f27414dc22fe5 be a valid fix

File: src/bin/lpython.cpp [hunk 3]
```diff
@@ -1658,7 +1658,9 @@ int main(int argc, char *argv[])
                     if (0 < err && err < 256) {
                         return err;
                     } else {
-                        return 1;
+                        // https://stackoverflow.com/a/27117435/15913193
+                        // https://linux.die.net/man/3/system
```
Comment (certik) [inline src/bin/lpython.cpp:1662]: Move these two links into the `utils.cpp` file to document `get_exit_status`?

**Review by certik — APPROVED** [review]
I think this looks good, I just left some minor comments.

File: src/lpython/utils.h [hunk 4]
```diff
@@ -12,6 +12,8 @@ std::string get_runtime_library_header_dir();
 bool is_directory(std::string path);
 bool path_exists(std::string path);
 
+int32_t get_exit_status(int32_t err);
```
Comment (certik) [inline src/lpython/utils.h:15]: Document here what the function does, you can put the links above here as well.

**Comment by Thirumalai-Shaktivel** [comment]
Thanks for the approval. Let's merge this.

**Pull request merged**